command/agent/command.go

```diff
@@ -596,6 +596,24 @@ WAIT:
 	}
 }
 
+func (c *Command) reloadHTTPServerOnConfigChange(newConfig *Config) error {
+	c.agent.logger.Println("[INFO] agent: Reloading HTTP server with new TLS configuration")
+	err := c.httpServer.Shutdown()
+	if err != nil {
+		return err
+	}
+
+	// Wait some time to ensure a clean shutdown
+	time.Sleep(5 * time.Second)
+	http, err := NewHTTPServer(c.agent, c.agent.config)
+	if err != nil {
+		return err
+	}
+	c.httpServer = http
```

Comment:
This assignment seems unsafe without a mutex around it, what if two different calls to this reload method happen close enough?

Reply:
Great catch, fixing.